hirte/src/common/parse-util.c

```diff
@@ -40,3 +42,128 @@ bool parse_port(const char *in, uint16_t *ret) {
         *ret = (uint16_t) l;
         return true;
 }
+
+bool is_IP(char *str) {
+        long num = 0;
+        int dots = 0;
+        char *ptr = NULL;
+
+        if (str == NULL) {
+                return false;
+        }
+
+        ptr = str;
+        // NOLINTNEXTLINE
+        while (*ptr != '\0') {
+                if (*ptr == '.') {
+                        dots++;
+                } else if (!isdigit(*ptr)) {
+                        return false;
+                }
+                ptr++;
+        }
+
+        if (dots != 3) {
+                return false;
+        }
+        int const max = 255;
+        ptr = strtok(str, ".");
+        int const base = 10;
+        while (ptr != NULL) {
+                num = strtol(ptr, NULL, base);
+                if (num >= 0 && num <= max) {
+                        ptr = strtok(NULL, ".");
+                } else {
+                        return false;
+                }
+        }
+
+        return true;
+}
+
+bool is_port(char *str) {
```

Comment:
We already have a function `parse_port`, which fails when the string does not contain a valid port. So no need to re-implement it.